### PR TITLE
fix: Refactor underlying methods to isolate calls to Path.cwd()

### PR DESCRIPTION
**Pull request by butler54**
Signed-off-by: Chris Butler <chris@thebutlers.me>

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] All commits are signed-off.

I need to write some new tests for the `trestle.utils.fs` module but otherwise this should be good to go. 

This PR isolates CWD calls which has the downside that cwd needs to be explicitly passed more, however, removes some bugs that impacted the `trestle.core.repository`

### CODE REVIEW

File: trestle/utils/fs.py [hunk 1]
```diff
@@ -93,30 +93,18 @@ def has_parent_path(sub_path: pathlib.Path, parent_path: pathlib.Path) -> bool:
...
 
-    if root_path is None or project_model_path is None:
-        raise err.TrestleError('Trestle project model not found')
+    Returns:
+        Type of Oscal Model for the provided model
+        Alias of that oscal model.
 
-    relative_path = path.relative_to(str(root_path))
+    """
     project_type = relative_path.parts[0]  # catalogs, profiles, etc
```
Comment (fsuits) [inline trestle/utils/fs.py:108]: I'm not sure how unhappy cases are handled.  What if the first part of path is foo?  Also what if you have catalogs/catalog.json?  i.e. valid object in wrong part of directory.

File: trestle/utils/load_distributed.py [hunk 2]
```diff
@@ -24,40 +24,40 @@
...
-def _load_list(file_path: Path) -> Tuple[Type[OscalBaseModel], str, List[OscalBaseModel]]:
+def _load_list(abs_path: Path, abs_trestle_root: Path) -> Tuple[Type[OscalBaseModel], str, List[OscalBaseModel]]:
     """Given path to a directory of list(array) models, load the distributed models."""
     aliases_not_to_be_stripped = []
     instances_to_be_merged: List[OscalBaseModel] = []
-    file_path = file_path.resolve()
-    collection_model_type, collection_model_alias = fs.get_stripped_contextual_model(file_path)
-
-    for path in sorted(Path.iterdir(file_path)):
+    abs_path = abs_path.resolve()
```
Comment (fsuits) [inline trestle/utils/load_distributed.py:31]: Kinda weird to call resolve on what is labeled an abs_path.  In that sense file_path is better.

File: trestle/utils/fs.py [hunk 3]
```diff
@@ -126,8 +114,7 @@ def get_contextual_model_type(path: pathlib.Path = None) -> Tuple[Type[OscalBase
     full_alias = model_alias
 
     for i in range(len(model_relative_path.parts)):
```
Comment (fsuits) [inline trestle/utils/fs.py:116]: I would rewrite this because of the range(len) and because the inner loop is somewhat opaque

**Review by fsuits — APPROVED** [review]
LGTM but some suggestions in comments for improvement.

File: trestle/utils/fs.py (hunk 1 above)
Comment (butler54) [reply trestle/utils/fs.py]: Now the project will choke if the incorrect directory is passed. This relative directory always relative to trestle project root directory.

File: trestle/utils/fs.py (hunk 3 above)
Comment (butler54) [reply trestle/utils/fs.py]: Cleaned up partially, however, it's inherited and a bit messy still. For not it is good enough for me.

**Pull request merged**